dpnp/dpnp_iface_arraycreation.py

```diff
@@ -901,16 +913,79 @@ def linspace(start, stop, num=50, endpoint=True, retstep=False, dtype=None, axis
 
     """
 
-    if not use_origin_backend():
-        if axis != 0:
-            checker_throw_value_error("linspace", "axis", axis, 0)
+    if retstep is not False:
+        pass
+    elif axis != 0:
+        pass
+    elif dpnp.isscalar(start) and dpnp.isscalar(stop):
+        usm_type = usm_type if usm_type is not None else "device"
+        dt = None if numpy.issubdtype(dtype, dpnp.integer) else dtype
+        res = dpnp_container.linspace(start,
+                                       stop,
+                                       num,
+                                       dtype=dt,
+                                       device=device,
+                                       endpoint=endpoint,
+                                       sycl_queue=sycl_queue,
+                                       usm_type=usm_type)
+        if numpy.issubdtype(dtype, dpnp.integer):
```

Comment:
As I got the reason of explicit calling of `dpnp.floor()` and setting `dt=None` for integer types while calling `dpt.linspace()` is because the example below gives different result in dpctl and numpy, isn't it?
```
numpy.linspace(-2.7, 12.8, 20, dtype=numpy.int64)
array([-3, -2, -2, -1,  0,  1,  2,  3,  3,  4,  5,  6,  7,  7,  8,  9, 10,
       11, 11, 12])

dpt.linspace(-2.7, 12.8, 20, dtype=numpy.int64)
usm_ndarray([-2, -1,  0,  0,  0,  1,  2,  3,  3,  4,  5,  6,  6,  7,  8,
              9,  9, 10, 11, 12])

dpnp.linspace(-2.7, 12.8, 20, dtype=numpy.int64)
[-3 -2 -2 -1  0  1  2  3  3  4  5  6  7  7  8  9 10 11 11 12]
```
Shouldn't dpctl behavior will be aligned with numpy instead?
Or if dpctl behavior is according to Python array API, shouldn't dpnp also follow it instead of aligning with numpy here?